lib/modules/manager/gomod/artifacts.ts

```diff
@@ -296,7 +298,21 @@ export async function updateArtifacts({
 
     const execCommands: string[] = [];
 
-    let args = 'get -d -t ./...';
+    let goGetDirs: string | undefined;
+    if (config.goGetDirs) {
+      goGetDirs = config.goGetDirs
+        .filter((dir) => {
+          const isValid = isValidLocalPath(dir);
+          if (!isValid) {
+            logger.warn({ dir }, 'Invalid path in goGetDirs');
+          }
+          return isValid;
+        })
+        .map(quote)
+        .join(' ');
+    }
+
+    let args = `get -d -t ${goGetDirs ?? './...'}`;
```

Comment:
should we check if `goGetDirs===' '` if the filter had returned an empty list?

Reply:
```
[].join(" ")
```
yields `""`

so that shouldn't be necessary

Reply:
but `(goGetDirs ?? './...') === ''`

Reply:
sure, I think that's okay.
if an user tries to specify some invalid godirs, it's better to let the command fail, than to "auto-magically fix it" to the default "./..."

Reply:
i would prefer to return an artifact error early to save time. so we don't install and run go if we know it fails anyways. we should not waste time

Reply:
would something like this suffice?
```
    if (goGetDirs === '') {
      throw new Error('Invalid goGetDirs');
    }
```